v1/api.go

```diff
@@ -678,6 +686,28 @@ func (q *QueryOpts) Ctx() context.Context {
 	return context.WithValue(context.Background(), contextKeyQueryOpts, q)
 }
 
+func MakeAPIError(values []reflect.Value) *APIError {
+	errIdx := len(values) - 1
+	if values[errIdx].IsNil() {
+		return nil
+	}
+
+	v := values[errIdx].Interface().(error)
```

Comment:
I recently learned that can get a second bool result from cast operations to see if they failed. I wonder if this would be useful here.